File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer>set=new HashSet<>();
        for(int num:nums){
            set.add(num);
        }
        int ans=0;
        for(int num:set){
          if(set.contains(num-1)){
            continue;
          }
          else{
            int l=1;
            while(set.contains(num+l)){
                l++;
            }
            ans=Math.max(ans,l);
          }
        }
        return ans;
    }
}

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int>seen;
        for(int num:nums){
            seen.insert(num);
        }
        int ans=0;
        for(int num:seen){
            if(seen.count(num-1)){
                continue;
            }
            else{
                int l=1;
                while(seen.count(num+l)){
                    l++;
                }
                   if(l>ans){
            ans=l;
         }
            }
      

        }
        return ans;
    }
};